backend/api/main.py

```diff
@@ -74,3 +86,35 @@ def delete_session(session=Depends(get_request_session)):
     session_id = session.get("session_id")
     destroy_session(session_id)
     del session["session_id"]
+
+
+@app.get("/user", dependencies=[Depends(verify_admin)])
+def get_users(db: Session = Depends(get_db)) -> list[models.GetUsersResponse]:
+    users = get_all_users(db)
+    return users
+
+
+@app.post("/user", dependencies=[Depends(verify_admin)])
+def create_user(
+    user: models.PostUserRequest, db: Session = Depends(get_db)
+) -> models.PostUserResponse:
+    new_user = create_db_user(db, user)
+    return new_user
+
+
+@app.put("/user/{id}", dependencies=[Depends(verify_admin)])
+def update_user(
+    user: models.PutUserRequest, db: Session = Depends(get_db)
+) -> models.PutUserResponse:
+    updated_user = update_db_user(db, user)
+    return updated_user
```

Comment:
Just as a note for our shared understanding: If a user's permission is modified, for the new permissions to take effect they'll need to logout and then login again. Not a big deal, but I thought I'd call that out (I guess a potential alternative would be to update permissions in any current sessions while updating the db, but that seems potentially dangerous + this isn't an operation we'd expect to do often so I think simpler is better).

Reply:
This is the case because the session state would still have the non-updated role values, correct? So in order to have updated role values, they'd need to logout and then login (thus creating a new session).

Reply:
Yep, exactly.